lib/http/features/raise_error.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module HTTP
+  module Features
+    class RaiseError < Feature
+      def initialize(ignore: [])
+        super()
+
+        @ignore = ignore
+      end
+
+      def wrap_response(response)
+        if response.code >= 400 && !@ignore.include?(response.code)
+          raise HTTP::StatusError, "Unexpected status code #{response.code}"
+        end
+
+        response
```

Comment:
WDYT about:


```suggestion
        return response if response.code < 400
        return response if @ignore.include?(response.code)
        
        raise HTTP::StatusError, response
```

Then changing `HTTP::StatusError` to:

``` ruby
   class StatusError < ResponseError
     attr_reader :response

     def initialize(response)
       @response = response

       super("Unexpected status code #{response.code}")
     end
   end
```